code generators/de.fhdo.lemma.model_processing.code_generation.container_base/src/main/java/de/fhdo/lemma/model_processing/code_generation/container_base/buildscript/OpendRootBuildScript.xtend

```diff
@@ -0,0 +1,96 @@
+package de.fhdo.lemma.model_processing.code_generation.container_base.buildscript
+
+import de.fhdo.lemma.model_processing.code_generation.container_base.template.BuildScriptTemplate
+import java.util.Map
+import java.util.HashMap
+import java.nio.file.Files
+import java.nio.file.Paths
+import java.io.File
+import java.nio.charset.StandardCharsets
+import java.util.regex.Pattern
+
+/**
+ * This class contains the logic for the generation of the build scripts, used in the
+ * [BuildScriptCodeGenerator].
+ *
+ * @author [Philip Wizenty](mailto:philip.wizenty@fh-dortmund.de)
+ */
+class OpendRootBuildScript {
+    static OpendRootBuildScript instance
+    Map<String, String> buildScriptNodes
+
+    new () {
+        this.buildScriptNodes = new HashMap
+    }
+
+    /*
+     * Get or create instance of the [OpendRootBuildScript] class.
+     */
+    static def OpendRootBuildScript getInstance() {
+        if (OpendRootBuildScript.instance === null)
+            OpendRootBuildScript.instance = new OpendRootBuildScript
+
+        return OpendRootBuildScript.instance
+    }
+
+    /*
+     * Load a  existing root build script from the file system into the data structure of the
+     * [OpendRootBuildScript] class.
+     * Note: The name of the nodes is extracted from the content of each line of the script via
+     * a regex. This may lead to a problem with build scripts not crated by this code generator.
+     */
+    def loadExistingRootBuildScript(String targetFolder) {
+        val file = new File(targetFolder + File.separator + "build" + File.separator + "build.sh")
+        if (file.exists) {
+            var stream = Files.lines(
+                Paths.get(targetFolder
```

Comment:
The argument of `Paths.get(...)` is the same as for `new File(...)`. Why not put it in a variable and reuse the variable here?